src/system_identity/input_data.rs

```diff
@@ -0,0 +1,31 @@
+//! The types contained inside this module are the required inputs for the System Identity creation
+//! supported methods.
+
+pub mod auth_method;
+pub mod environment;
+pub mod output_platform;
+
+use auth_method::AuthMethod;
+use environment::NewRelicEnvironment;
+use output_platform::OutputPlatform;
+
+/// Represents the input data required to create a System Identity.
+///
+/// The rationaly behind this struct is to group all the necessary information, so any consumer of
+/// the System Identity creation functionality should make sure it can create this object,
+/// for example from CLI arguments or by library usage, before reaching for any actual creation service.
+#[derive(Debug, Clone, PartialEq)]
+pub struct SystemIdentityCreationMetadata {
+    pub system_identity_input: SystemIdentityInput,
+    pub name: String,
+    pub environment: NewRelicEnvironment,
+    pub output_platform: OutputPlatform,
+}
+
+/// The existing System Identity data that is required as input to create a new System Identity.
```

Comment:
This is not needed to "create" but to "authenticate", right?
I can authenticate with an L1 but create an L2. The data changes.

For instance, creating an L2 requires sending the public key https://newrelic.atlassian.net/wiki/spaces/INST/pages/4315907146/AC+Deployment+Scale+Create+L2+to+create+L2#Create-the-child-identity-using-the-API-%2B-Token-(not-NerdGraph). Authenticating with an L2 requires sending a signed JWT token.